indexer/elastic.py

```diff
@@ -44,3 +52,23 @@ def elasticsearch_client(self) -> Elasticsearch:
 
         # Connects immediately, performs failover and retries
         return Elasticsearch(hosts.split(","))
+
+    def read_file(self, template_name: str) -> Union[dict, Any]:
+        assert self.args
+        elasticsearch_config_dir = self.args.elasticsearch_config_dir
+        file_path = os.path.join(elasticsearch_config_dir, template_name)
+        with open(file_path, "r") as file:
+            data = file.read()
+        return json.loads(data)
```

Comment:
```suggestion
    def _load_template(self, name: str) -> dict | Any:
        file_path = os.path.join(self.elasticsearch_config_dir, name)
        with open(file_path, "r") as file:
            data = file.read()
        return json.loads(data)
```

1. read_file is too generic
2. I'm assuming this method is supposed to be private-ish
3. Unless there are other benefits, new code should [prefer](https://peps.python.org/pep-0604/) `|` over `Union`